plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/ElkLayered.java

```diff
@@ -341,12 +341,11 @@ private void reviewAndCorrectHierarchicalProcessors(final LGraph root, final Col
             });
         }
         
-        // Greedy switch
-        //  simply turn it off for children
-        if (!GraphConfigurator.activateGreedySwitchFor(root)) {
-            graphs.forEach(
-                    g -> g.setProperty(LayeredOptions.CROSSING_MINIMIZATION_GREEDY_SWITCH_TYPE, GreedySwitchType.OFF));
-        }
+        // Greedy switch (simply copy the behavior of the root to all children)
```

Comment:
From looking at the old implementation and how crossing minimization itself is configured, I feel that every hierarchy level requires the hierarchical-aware layout processor to be configured (even though it won't get executed). 

I didn't check this thoroughly though.

Reply:
Well, that would make sense to me.